src/App.jsx

```diff
@@ -14,45 +16,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
+
+function sortGoods(goods, { sortField, isSorted }) {
+  const localGoods = [...goods];
+
+  if (sortField) {
+    localGoods.sort((good1, good2) => {
+      switch (sortField) {
+        default:
+          return 0;
+
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+      }
+    });
+  }
+
+  if (isSorted) {
+    localGoods.reverse();
+  }
+
+  return localGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isSorted, setSorted] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const showGoods = sortGoods(
+    goodsFromServer,
+    { sortField, isSorted },
+  );
+
+  const reset = () => {
+    setSortField('');
+    setSorted(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(
+            ['button', 'is-info'],
+            { 'is-light': sortField !== SORT_FIELD_NAME },
+          )}
+          onClick={() => setSortField(SORT_FIELD_NAME)}
```

Comment:
it is bas practise to make anonymous arrow functions in jsx
consider to make function factory which will accept field and return handler.
const makeSetSortField = (field) => () => setSortField(field);

```suggestion
          onClick={makeSetSortField(SORT_FIELD_NAME)}
```